script/tool/test/publish_check_command_test.dart

```diff
@@ -132,6 +132,26 @@ void main() {
       expect(runner.run(<String>['publish-check']), throwsA(isA<ToolExit>()));
     });
 
+    test('Success message on stderr is not printed as an error', () async {
+      createFakePlugin('d');
+
+      const String publishOutput = 'Package has 0 warnings.';
+
+      final MockProcess process = MockProcess();
+      process.stderrController.add(publishOutput.codeUnits);
+      process.stdoutController.close(); // ignore: unawaited_futures
+      process.stderrController.close(); // ignore: unawaited_futures
+
+      process.exitCodeCompleter.complete(0);
+
+      processRunner.processesToReturn.add(process);
+
+      final List<String> output = await runCapturingPrint(
+          runner, <String>['publish-check']);
+
+      expect(output, isNot(contains(contains('ERROR:'))));
```

Comment:
what is the behavior of `contains(contains(` is this case?

Reply:
The outer `contains` is "Does this list contain an element that satisfies the given matcher?" and the inner `contains` is "Does this string contain the substring "ERROR:". So this is expecting that there is no line of output with "ERROR:" as a substring.

Reply:
ah - interesting